### PR TITLE
Document that only pod template is considered for triggering a deployment new revision

### PR DESCRIPTION
**Pull request by janetkuo**
Ref https://github.com/kubernetes/kubernetes/issues/25236#issuecomment-217995389

cc @bgrant0607 @kubernetes/docs

### CODE REVIEW

File: docs/user-guide/deployments.md [hunk 1]
```diff
@@ -225,7 +229,11 @@ before changing course.
 Sometimes you may want to rollback a Deployment; for example, when the Deployment is not stable, such as crash looping.
 By default, all of the Deployment's rollout history is kept in the system so that you can rollback anytime you want 
 (you can change that by specifying [revision history limit](/docs/user-guide/deployments/#revision-history-limit)). 
-You should note that only the update to the Deployment's "pod template" is seen as a different revision of rollout.
+
+**Note:** a Deployment's revision is created when a Deployment's rollout is triggered. This means that the new revision is created 
+if and only if the Deployment's pod template (i.e. `.spec.template`) is changed, e.g. updating labels or container images of the template labels.
```
Comment (bgrant0607) [inline docs/user-guide/deployments.md]: Typo: images of the template

File: docs/user-guide/deployments.md [hunk 2]
```diff
@@ -225,7 +229,11 @@ before changing course.
 Sometimes you may want to rollback a Deployment; for example, when the Deployment is not stable, such as crash looping.
 By default, all of the Deployment's rollout history is kept in the system so that you can rollback anytime you want 
 (you can change that by specifying [revision history limit](/docs/user-guide/deployments/#revision-history-limit)). 
-You should note that only the update to the Deployment's "pod template" is seen as a different revision of rollout.
+
+**Note:** a Deployment's revision is created when a Deployment's rollout is triggered. This means that the new revision is created 
+if and only if the Deployment's pod template (i.e. `.spec.template`) is changed, e.g. updating labels or container images of the template labels. 
+Other updates, such as scaling the Deployment, will not create a Deployment revision. This implies that when you rollback to an earlier revision, 
+only the Deployment's pod template part will be rolled back.
```
Comment (bgrant0607) [inline docs/user-guide/deployments.md]: A little explanation of why this is (to facilitate simultaneous manual or auto-scaling) would be helpful.

**Comment by bgrant0607** [comment]
Thanks. Couple comments.

**Comment by bgrant0607** [comment]
This is pretty small. Feel free to squash when you push changes.

File: docs/user-guide/deployments.md (hunk 1 above)
Comment (janetkuo) [inline docs/user-guide/deployments.md]: fixed

File: docs/user-guide/deployments.md (hunk 2 above)
Comment (janetkuo) [inline docs/user-guide/deployments.md]: done

**Comment by janetkuo** [comment]
All comments addressed @bgrant0607

**Comment by bgrant0607** [comment]
LGTM, thanks

**Pull request merged**